sources/web/datalab/auth.ts

```diff
@@ -136,7 +138,10 @@ export function handleAuthFlow(request: http.ServerRequest, response: http.Serve
           logging.getLogger().info('Got tokens');
           oauth2Client.setCredentials(tokens);
           // Push them to Jupyter and handle request.
-          persistCredentials(tokens);
+          var email = persistCredentials(tokens);
+          if (email) {
+            settings.instanceUser = email;
```

Comment:
I think it is probably fine to not set instanceUser so it is consistently "anonymous" in all cases.